packages/aws-cdk-lib/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -325,14 +325,21 @@ export class HostedRotation implements ec2.IConnectable {
       this.masterSecret.denyAccountRootDelete();
     }
 
+    let masterSecretArn: string | undefined;
+    if (this.masterSecret?.secretFullArn) {
+      masterSecretArn = this.masterSecret.secretArn;
+    } else if (this.masterSecret) { // ISecret as an imported secret with partial ARN
+      masterSecretArn = this.masterSecret.secretArn + '-??????';
+    }
```

Comment:
The case where `secretArn` and `secretName` are Token is also taken into account here.

- `if (this.masterSecret?.secretFullArn) {`
  - If we create a secret construct in the stack, here is passed.
  - Otherwise `this.masterSecret` is from `fromSecretArn`, `fromSecretCompleteArn` methods, etc...
    - This `if` statement will be true even if the arg is a token or not.

Reply:
I'll think about this some more, so I'll draft it once.